src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -139,6 +139,8 @@ public sealed partial class NpgsqlConnector : IDisposable
 
         internal PreparedStatementManager PreparedStatementManager;
 
+        internal SqlQueryParser SqlQueryParser { get; } = new();
```

Comment:
Only create if used?
```suggestion
        private SqlQueryParser? _sqlQueryParser;
        internal SqlQueryParser SqlQueryParser => _sqlQueryParser ??= new();
```

Reply:
It's still used for execution before commands become prepared, so basically always...

Also, note that I moved it to NpgsqlConnector, which means every physical just has its own instance which it reuses. So even when used, it's still better than before where we instantiated on command execution.

Reply:
Ah so its 1:N of `NpgsqlConnection` : `NpgsqlConnector`?

Reply:
Yep. NpgsqlConnector is the physical connection - the thing that gets pooled internally inside Npgsql, whereas NpgsqlConnection is basically just a thin façade over it. So NpgsqlConnector holds lots of "heavy resources", but that's OK since everything naturally gets recycled via pooling.

(BTW I have some ideas about how to also recycle NpgsqlConnection, NpgsqlCommand and maybe even parameter instances, for .NET 7. The EF benchmarks actually already do that - EF has its own pooling of DbContexts, each of which has its own NpgsqlConnection, which has its own NpgsqlCommand. So things are pooled/recycled all the way up)